vendor/engines/bulk_email/app/services/bulk_email/bulk_email_searcher.rb

```diff
@@ -7,15 +7,22 @@ class BulkEmailSearcher
     attr_reader :order_details, :search_fields
 
     DEFAULT_SORT = [:last_name, :first_name].freeze
-    SEARCH_TYPES = [:customers, :account_owners, :customers_and_account_owners, :authorized_users].freeze
+    USER_TYPES = %i(customers authorized_users account_owners).freeze
 
     def initialize(search_fields)
       @search_fields = search_fields
     end
 
+    def user_types
+      @user_types ||=
+        USER_TYPES & (search_fields[:bulk_email].try(:[], :user_types) || []).map(&:to_sym)
+    end
+
     def do_search
-      return unless SEARCH_TYPES.include? search_fields[:search_type].to_sym
-      public_send(:"search_#{search_fields[:search_type]}")
+      return if user_types.blank?
```

Comment:
…which also means if a user deselects all checkboxes and does a search, it will *not* say "No users found" but probably should (or something more specific, like "Choose at least one user type").